### ISSUE TITLE
THREAD - Multiple space heroes spawn their normal hero too, keeping things permanently contested

### ISSUE THREAD

**Issue by AlyMar1994** [AlyMar1994/UFoCP#146]
_This issue is extremely similar to https://github.com/AlyMar1994/UFoCP/issues/27, https://github.com/AlyMar1994/UFoCP/issues/28 & https://github.com/AlyMar1994/UFoCP/issues/91_

Space heroes that also have an attached land unit will permanently contest capturable objects should they spawn next to them.  This is because the game spawns the unit alongside the actual hero unit.  Petroglyph was aware of this to some degree, as they have specific code preventing Bossk's hero icon from appearing when he betrays Tyber Zann and despawns from the mission:

```
-- BOSSK
	loc_start = Find_Hint("STORY_TRIGGER_ZONE", "bosskpos")
	bossk = Find_First_Object("HOUNDSTOOTH")
	if TestValid(bossk) then
		bossk.Despawn()
		
		--Need to despawn Bossk himself also to eliminate hero icon.
		bossk_land_version = Find_First_Object("BOSSK")
		if TestValid(bossk_land_version) then
			bossk_land_version.Despawn()
		end
	end
```

Generally, this bug isn't solvable for GC as heroes need to be linked to their space units.  However, we can fix this for skirmish by creating special skirmish-only duplicates of the space heroes and using them instead of the normal heroes. Skirmish AI in TacticalMultiplayerBuildSpaceUnitsGeneric needs to be updated. A full list of bug report threads that involve the indicator being unused will be kept here.  This bug report will be kept open until all portions of mishandling in the code are resolved.

**Related Issues:**
- https://github.com/AlyMar1994/UFoCP/issues/147
- https://github.com/AlyMar1994/UFoCP/issues/148
- https://github.com/AlyMar1994/UFoCP/issues/149
- https://github.com/AlyMar1994/UFoCP/issues/150
- https://github.com/AlyMar1994/UFoCP/issues/151
- https://github.com/AlyMar1994/UFoCP/issues/152
- https://github.com/AlyMar1994/UFoCP/issues/153

**Comment by AlyMar1994** [comment]
Hacky, but it might be worthwhile adding a Lua script to these units and deleting their hero like how the campaign does when they spawn.

**Comment by AlyMar1994** [comment]
Closing as this was fixed in FoC version 1.121.859328 on 10/2024. These changes will need to be reverted in the future.
